### ISSUE TITLE
Managing focus: aria-activedescendant and roving tabs

### ISSUE THREAD

**Issue by adamsilver** [adamsilver/f#175]
The first is to use the `aria-activedescendant` attribute. The way the attribute works is to always keep focus on the component's container, but tell the container which element within is active. This doesn't work for the autocomplete control because the text box which will be in focus in order to type into it, is a sibling of the menu, not the parent.

The second approach is to use the roving tabindex technique. When using roving tabindex to manage focus in a composite UI component, the element that's to be included in the tab sequence has tabindex of "0" and all other focusable elements contained in the composite have tabindex of "-1". In our case, this would mean all the options in the menu would have a tabindex=”-1”

We're going to be using the roving tabindex, but not exactly as described above. This is because the text box is focusable by default. Rather than mess with it's tabindex, we can listen for when the user leaves the field using JavaScript. In which case, we just need to hide the menu and focus the next focusable element—which if using the Tab key, browsers will do by default.
